src/main/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServer.scala

```diff
@@ -102,12 +102,39 @@ trait JsonRpcHttpServer extends Json4sSupport {
     complete(httpResponseF)
   }
 
-  private def handleRequest(request: JsonRpcRequest) = {
-    complete(jsonRpcController.handleRequest(request))
+  private def handleRequest(request: JsonRpcRequest) = extractClientIP { ip =>
+    complete {
+      val requestId = request.id.flatMap(_.extractOpt[String]).getOrElse("n/a")
+
+      log.info(s"Processing new JSON RPC request [$requestId] from [$ip]:\n ${serialization.write(request)}")
+
+      val responseF = jsonRpcController.handleRequest(request)
+      responseF.onComplete {
```

Comment:
Minor, we could use `andThen` to both signify that we have an effect and reuse the returned value, which is the original future.